fasthtml/core.py

```diff
@@ -191,6 +191,10 @@ async def _find_p(req, arg:str, p:Parameter):
     if (res in (empty,None)) and p.default is empty: raise HTTPException(400, f"Missing required field: {arg}")
     # If we have a default, return that if we have no value
     if res in (empty,None): res = p.default
+    # if anno is a Generic Alias but res isn't an iterable then wrap in list
+    if isinstance(anno, GenericAlias):
```

Comment:
I think there are parameterised generics that aren't lists? We should probably me more specific about what we're looking for.

Reply:
@jph00 The challenge is that for this error `anno` is a `GenericAlias`, which arguably is a bug in its own right. Or maybe we're just not particular enough so this is a missing feature? Looking into that now.

In the meantime, I've changed the `try/except` to a more precise check. Hopefully that goes away, replaced with a more accurate `anno` specification.

Reply:
Diving into Fastcore's [signature_ex](https://fastcore.fast.ai/basics.html#signature_ex) function.

Reply:
Hmmm... I think `signature_ex` is a tangent. Digging into the behavior around params in `_wrap_req`.